src/components/common/SearchInput/SearchInput.tsx

```diff
@@ -40,26 +64,67 @@ const SearchInput = React.forwardRef<SearchInputRef, SearchInputProps>(
     { onSearchClick, buttonTestId = 'store-search-button', ...props },
     ref
   ) {
+    const [searchQuery, setSearchQuery] = useState<string>('')
+    const [suggestionsOpen, setSuggestionsOpen] = useState<boolean>(false)
+    const [suggestions, setSuggestions] =
+      useState<SearchSuggestionsQueryQuery['search']['suggestions']>()
+
     const { addToSearchHistory } = useSearchHistory()
+    const searchRef = useRef<HTMLDivElement>(null)
     const handleSearch = (term: string) => {
       addToSearchHistory(term)
       doSearch(term)
+      setSuggestionsOpen(false)
+      setSearchQuery(term)
+    }
+
+    useOnClickOutside(searchRef, () => setSuggestionsOpen(false))
+
+    const handleChange: ChangeEventHandler<HTMLInputElement> = async (
+      event
+    ) => {
+      setSearchQuery(event.target.value)
+
+      if (event.target.value) {
+        const { search } = await request<
+          SearchSuggestionsQueryQuery,
+          SearchSuggestionsQueryQueryVariables
+        >(SearchSuggestionsQuery, {
+          term: event.target.value,
+        })
+
+        setSuggestions(search.suggestions)
+      }
```

Comment:
Have you noticed if this request sequenced by set states creates a long task?

Reply:
I don't understand what do you say

Reply:
Ok, let me try to explain better. This event handler does basically 3 things: setSearchQuery, awaits a Promise, then setSuggestion. The useStorage, had problems creating a long task, with a similar task sequence: react effect, promise, setState.
So, I asked you, to profile this click action, to ensure if this callback creates a long task.

Reply:
When i was trying to profile i noticed this bug
![image](https://user-images.githubusercontent.com/15680320/161252811-a4d0681a-fffe-4e44-8768-6607cbf40dbc.png)